### PR TITLE
Issue 83 revamp backend

### PR DESCRIPTION
**Pull request by jvaughn575**
Took a little bit of working through but I think I improved the build process slightly for the openshift build. Now using environmental variables we can also set the Allow-control-access-origin headers to only accept requests from our client. I will publish a list of env variables to set when deploying the server soon.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -13,80 +13,89 @@ import {InspirationRouter} from './routes/inspirationRouter';
...
-    /******* Change below line to restrict to our website origin *********/    
-    res.header('Access-Control-Allow-Origin', req.headers.origin);    
-    res.header('Access-Control-Allow-Methods', 'GET,PUT,POST,DELETE,PATCH');
-    res.header('Access-Control-Allow-Headers', 'Content-Type, Authorization');
-    next();
-   });
+    // Passport requirements
+    // Not secure must change before production
+    app.use(session({ 
+      secret: env.EXPRESS_SECRET || 'jk_travelhive',
```
Comment (reggieroby) [inline server/server.js]: thats a nice addition.

File: config/config.js [hunk 2]
```diff
@@ -1,31 +1,52 @@
...
+        "logging" : false,
+      },
+      "environment": env
+    }
+    break;
+
+    case "production":
+      Config.Database = {
+        "name": process.env.MYSQL_DBNAME,
+        "user": process.env.MYSQL_USER,
```
Comment (reggieroby) [inline config/config.js]: add to this process to the readme.

File: package.json [hunk 3]
```diff
@@ -35,6 +35,8 @@
     "server": "cross-env NODE_ENV=development nodemon --exec babel-node ./server/server.js",
     "chat:server": "cross-env NODE_ENV=development nodemon --exec babel-node ./server/sideKickChatServer.js",
     "build": "react-scripts build",
+    "build:server": "babel server -d build/serverDist/server && babel models -d build/serverDist/models && babel config -d build/serverDist/config && babel migrations -d build/serverDist/migrations",
+    "server:openshift:deploy":"git subtree push --prefix build/serverDist origin openshift_server_deploy",
```
Comment (reggieroby) [inline package.json]: you should add what your doing here to the readme.

File: server/server.js [hunk 4]
```diff
@@ -13,80 +13,89 @@ import {InspirationRouter} from './routes/inspirationRouter';
 
 export const app = express();
 export const httpServer = require('http').createServer(app);
-const port = process.env.PORT || 3001;
+
+/* 8080 Port is the default for Openshift deployment */
+const port = process.env.PORT || process.env.OPENSHIFT_NODEJS_PORT || '3001',
```
Comment (reggieroby) [inline server/server.js]: the diff between using the node env vars and the openshift vars as well?

**Comment by reggieroby** [comment]
looks good. id add some docs to the readme, so others can replicate and know what all your doing.

File: server/server.js (hunk 4 above)
Comment (jvaughn575) [reply server/server.js]: Possible not. Its just if there's a problem I can set that openshift environment variable in the openshift web console.

I had a lot of trouble getting some stuff to work, so the end result is somethings may not make sense. I am still new to a lot of this, so I'm making lots of mistakes.

**Pull request merged**